### DISCUSSION TITLE
VS Code not suggesting Textual auto imports

### DISCUSSION THREAD

**Discussion by denisrosset** [Textualize/textual#2111] [Ideas]
I've checked the issues so far.

My issue is that VS Code does not offer imports for standard textual identifiers such as Vertical.

The culprit is the analysis depth of the Pylance extension. This can be configured by setting the "python.analysis.packageIndexDepths" to 10 (which is probably overkill).

I suggest to document this necessary setting somewhere in the documentation, for ex. "Textual and IDEs", where the peculiarities of the VS Code Terminal emulation could as well be discussed.

**Comment by msafi** [comment]
This was a tricky issue to fix. Thanks for the solution!
